test/nlp_program.jl

```diff
@@ -627,6 +632,125 @@ function test_compute_derivatives_Finite_Diff(;
     end
 end
 
+################################################
+#=
+# Test Objective Sensitivity wrt Parameters
+=#
+################################################
+
+function test_ObjectiveSensitivity()
```

Comment:
this is only testing the case with parameters in the RHS which is the one that can rely on duals.

we need some teste for parameters in the middle of nonlinear function.

my suggestion is a test with parameter * variable and another one in a non-linear term.

each of these test should have analystical solutions.

It would be nice to have a simple test where we have 10 parameters but we only differentiate with respect to the 3rd and 7th for instance. to test the sparsity handling.